### PR TITLE
feat: add samples and tests for adding captions to a job

### PR DESCRIPTION
**Pull request by irataxy**
Needs https://cloud.google.com/transcoder/images/caption.srt uploaded to the test bucket for the tests to pass.

Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [X] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-video-transcoder/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [X] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes 210892114

### CODE REVIEW

**Comment by busunkim96** [comment]
Hey @irataxy, do you have an AI DPE helping you with getting that new file to the bucket or do you need someone else to look into that?

**Comment by irataxy** [comment]
> Hey @irataxy, do you have an AI DPE helping you with getting that new file to the bucket or do you need someone else to look into that?

@nicain added it so this should be ready for review.

File: samples/snippets/create_job_with_embedded_captions.py [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+            ),
+            transcoder_v1.types.ElementaryStream(
+                key="audio-stream0",
+                audio_stream=transcoder_v1.types.AudioStream(
+                    codec="aac", bitrate_bps=64000
+                ),
+            ),
+            transcoder_v1.types.ElementaryStream(
+                key="cea-stream0",
+                # The following doesn't work because "mapping" is a reserved
```
Comment (busunkim96) [inline samples/snippets/create_job_with_embedded_captions.py:95]: Thanks for explaining this in the samples!

File: samples/snippets/create_job_with_embedded_captions.py [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+                file_name="manifest.m3u8",
+                # In a similar situation to `mapping`, the autogenerated
+                # library added _ to the end of the reserved word `type`.
+                type_="HLS",
+                mux_streams=["sd-hls"],
+            ),
+            transcoder_v1.types.Manifest(
+                file_name="manifest.mpd",
+                # In a similar situation to `mapping`, the autogenerated
+                # library added _ to the end of the reserved word `type`.
```
Comment (busunkim96) [inline samples/snippets/create_job_with_embedded_captions.py:155]: I think it's okay to omit this explanation here since it's close to the one above.

**Review by busunkim96 — APPROVED** [review]

File: samples/snippets/create_job_with_embedded_captions.py (hunk 2 above)
Comment (irataxy) [reply samples/snippets/create_job_with_embedded_captions.py]: Done

File: samples/snippets/create_job_with_embedded_captions.py (hunk 1 above)
Comment (irataxy) [reply samples/snippets/create_job_with_embedded_captions.py:95]: No problem

**Pull request merged**